services/doctors/crud/agenda.py

```diff
@@ -49,4 +49,13 @@ def group_by_date(self,agenda_list: list):
                     }
                 ]
 
-        return agenda_final
\ No newline at end of file
+        return agenda_final
+
+    def delete(self,agenda_id: int, doctor_id: int):
+        
+        self.db.query(Agenda).filter(
+            Agenda.id == agenda_id,
+            Agenda.doctor_id == doctor_id
+            ).delete()
+        self.db.commit()
```

Comment:
**Suggestion:** Consider handling the case when the delete operation fails. You can use SQLAlchemy's `delete` method return value to check if any rows were actually deleted. If no rows were deleted, it means the `agenda_id` or `doctor_id` was not found in the database.
```suggestion
    def delete(self,agenda_id: int, doctor_id: int):

        deleted_rows = self.db.query(Agenda).filter(
            Agenda.id == agenda_id,
            Agenda.doctor_id == doctor_id
            ).delete()
        self.db.commit()
        if deleted_rows == 0:
            raise ValueError('No agenda found with the provided id and doctor id')
        return None
```